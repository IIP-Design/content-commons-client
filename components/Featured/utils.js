import { capitalizeFirst } from 'lib/utils';
import { normalizeItem } from 'lib/elastic/parser';
import { postTypeAggRequest } from 'lib/elastic/api';

/**
 * Get a string of up to three categories from a content item
 * @param {Object} item Normalized data for a give content item
 * @returns {string} Lowercase list of category names separated by ·
 */
export const getCategories = item => {
  const cats = item?.categories?.slice( 0, 3 ).reduce( ( acc, cat, index, arr ) => {
    const c = acc + cat.name.toLowerCase();

    return index < arr.length - 1 && index < 2 ? `${c} · ` : c;
  }, '' );

  return cats;
};

/**
 * Updates the the featured context
 * @param {Promise[]} array Group of promises that resolve the recents properties
 * @param {dispatch} dispatch A reducer dispatch function
 */
export const getFeatured = async ( array, dispatch ) => {
  const resArr = await Promise.all( array ).catch( err => {
    dispatch( {
      type: 'LOAD_FEATURED_FAILED',
    } );
  } );

  console.dir( resArr );

  const priorities = {};
  const recents = {};
  let items;

  if ( resArr ) {
    resArr.forEach( res => {
      if ( res?.data?.hits?.hits ) {
        items = res.data.hits.hits.map( item => normalizeItem( item, res.locale ) );

        switch ( res.component ) {
          case 'priorities':
            priorities[res.term] = items;
            break;
          case 'packages':
          case 'recents':
            recents[res.postType] = items;
            break;
          default:
            break;
        }
      }
    } );

    dispatch( {
      type: 'LOAD_FEATURED_SUCCESS',
      payload: {
        priorities,
        recents,
      },
    } );
  }
};


/**
 *
 * @param {dispatch} dispatch A reducer dispatch function
 * @param {Object} user User date object
 */
export const loadPostTypes = async ( dispatch, user ) => {
  dispatch( { type: 'LOAD_POST_TYPES_PENDING' } );

  let response;

  try {
    response = await postTypeAggRequest( user );
  } catch ( err ) {
    dispatch( { type: 'LOAD_POST_TYPES_FAILED' } );

    return;
  }

  if ( response?.aggregations?.postType?.buckets ) {
    const { buckets } = response.aggregations.postType;

    const filtered = buckets.filter( type => type.key !== 'courses' && type.key !== 'page' && type.key !== 'package' );

    const payload = filtered.map( type => {
      let displayName = capitalizeFirst( type.key );

      if ( type.key === 'post' ) displayName = 'Article';
      if ( type.key === 'document' ) displayName = 'Press Releases and Guidance';

      return {
        key: type.key,
        display_name: displayName,
        count: type.doc_count,
      };
    } );

    dispatch( {
      type: 'LOAD_POST_TYPES_SUCCESS',
      payload,
    } );
  } else {
    dispatch( {
      type: 'LOAD_POST_TYPES_SUCCESS',
      payload: [],
    } );
  }
};
