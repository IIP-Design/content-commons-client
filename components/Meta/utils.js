/**
 * Check if content exists for a given meta property
 * If not, filters it from the meta object
 *
 * @param {Object} meta All possible meta tags
 * @returns {Object} Only those meta tags for which there is content
 */
const filterEmpty = meta => {
  const filteredMeta = {};
  const metaArr = Object.entries( meta );

  metaArr.forEach( item => {
    if ( item[1] ) {
      const property = item[0];
      const value = item[1];

      filteredMeta[property] = value;
    }
  } );

  return filteredMeta;
};

/**
 * Converts the content item object into a metavalues array using the checkAndPush method
 *
 * @param {object} item Object of metadata for a given content item
 * @param {string} url The URL of the page to be rendered
 * @returns {array} Array of { property: content } key pairs
 */
export const getOgTags = ( item, url ) => {
  const {
    selectedLanguageUnit,
    title,
    type,
  } = item;

  const isVideo = type && type === 'video';

  /* TEMP vars for Document use case */
  const trimContent = item.content && item.content.rawText && `${item.content.rawText.substring( 0, 152 )}...`;
  const description = item.description || trimContent;

  const videoSrc = selectedLanguageUnit && selectedLanguageUnit.source && selectedLanguageUnit.source[0]
    ? selectedLanguageUnit.source[0]
    : {};
  const streamSrc = videoSrc.stream && videoSrc.stream.link ? videoSrc.stream.link : '';

  const meta = {
    'og:type': isVideo ? 'video.other' : 'article',
    'og:url': url,
    'twitter:card': isVideo ? 'player' : 'summary',
    'og:description': description,
    'og:title': title,
    'og:video:url': isVideo ? streamSrc : '',
    'twitter:player': isVideo ? streamSrc : '',
    'twitter:player:height': isVideo ? '196px' : '',
    'twitter:player:width': isVideo ? '350px' : '',
  };

  return filterEmpty( meta );
};
