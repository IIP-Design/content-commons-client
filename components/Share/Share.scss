@import "../../styles/colors.scss";

.popup_share {
  min-width: 540px;
  min-height: 360px;
  padding: 1em;
  background-color: #fff;

  .ui.header {
    // margin: 0;
    // font-size: 1.2em;
    // font-weight: bold;
    // color: $medium-grey;
    margin-bottom: .77em;
    font-size: 1.4em;
    font-weight: 400;
    color: $blue;
  }
}

.ui.list.share_list {
  margin-top: 2.5em;

  a.item:hover .icon {
    color: #fff;
    transform: scale(1.2);
    transition: transform 0.2s;
  }

  .item {
    margin-bottom: 1em;

    .icon {
      position: relative;
      padding: 15px;
      border-radius: 50%;
      color: #fff;
      top: -5px;

      &::before {
        position: absolute;
        top: 50%;
        right: 50%;
        transform: translate(50%, -50%);
      }

      &.facebook {
        background-color: $fb-blue;
      }

      &.twitter {
        background-color: $twitter-blue;
      }
    }

    .content {
      margin-left: 5px;
      font-size: 1em;
      color: $medium-grey;
    }
  }
}
