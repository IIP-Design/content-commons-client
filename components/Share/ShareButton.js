import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';
import { openWindow, isMobile } from 'lib/browser';

const ShareButton = props => {
  const {
    url, icon, isPreview, label,
  } = props;

  /**
   * Opens a new window. If mobile, go a new window, else open popup window
   * Calling this function as opposed to directly calling openWindow allows
   * us to fetch window config if present on the element. It also does not force
   * deaking with the event object in the openWindow method
   * @param {*} e event object
   * @param {*} data element data attributes
   */
  const willOpenWindow = ( e, data = {} ) => {
    if ( isPreview ) {
      e.preventDefault();

      return;
    }

    if ( !isMobile() ) {
      e.preventDefault();
      openWindow( url, data );
    }
  };

  return (
    <List.Item
      as="a"
      href={ url }
      rel="noopener noreferrer"
      target="_blank"
      onClick={ willOpenWindow }
      onKeyPress={ willOpenWindow }
      style={ { cursor: isPreview ? 'default' : 'pointer' } }
    >
      <List.Icon name={ icon } />
      <List.Content>{ label }</List.Content>
    </List.Item>
  );
};

ShareButton.propTypes = {
  url: PropTypes.string,
  icon: PropTypes.string,
  isPreview: PropTypes.bool,
  label: PropTypes.string,
};

export default ShareButton;
