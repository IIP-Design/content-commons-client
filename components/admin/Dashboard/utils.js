import { getProjectsType } from 'lib/graphql/util';
import {
  DELETE_GRAPHIC_PROJECT_MUTATION,
  TEAM_GRAPHIC_PROJECTS_QUERY,
  TEAM_GRAPHIC_PROJECTS_COUNT_QUERY,
  GRAPHIC_PROJECT_IMAGE_FILES_QUERY,
  UNPUBLISH_GRAPHIC_PROJECT_MUTATION,
  UPDATE_GRAPHIC_STATUS_MUTATION,
  GRAPHIC_PROJECTS_META_QUERY,
} from 'lib/graphql/queries/graphic';
import {
  DELETE_PACKAGE_MUTATION,
  PACKAGE_FILES_QUERY,
  TEAM_PACKAGES_COUNT_QUERY,
  TEAM_PACKAGES_QUERY,
  UNPUBLISH_PACKAGE_MUTATION,
  UPDATE_PACKAGE_STATUS_MUTATION,
  PACKAGES_META_QUERY,
} from 'lib/graphql/queries/package';
import {
  DELETE_VIDEO_PROJECT_MUTATION,
  TEAM_VIDEO_PROJECTS_COUNT_QUERY,
  TEAM_VIDEO_PROJECTS_QUERY,
  UNPUBLISH_VIDEO_PROJECT_MUTATION,
  UPDATE_VIDEO_STATUS_MUTATION,
  VIDEO_PROJECT_FILES_QUERY,
  VIDEO_PROJECTS_META_QUERY,
} from 'lib/graphql/queries/video';
import { TEAM_PLAYBOOKS_QUERY, TEAM_PLAYBOOKS_COUNT_QUERY } from 'lib/graphql/queries/playbook';

/**
 * Returns the project type based off of the content type available to a team
 *
 * @param {Object} team team data received from GraphQL
 */
export const getContentTypesFromTeam = team => {
  const contentTypes = team?.contentTypes ? team.contentTypes : '';

  const type = getProjectsType( contentTypes );

  return type || null;
};

/**
 * Returns the expected queries depending on what content types a team has access to
 *
 * @param {Object} team team data received from GraphQL
 * @returns {Object} list of relevant queries
 */
export const setQueries = team => {
  const queries = {};

  switch ( getContentTypesFromTeam( team ) ) {
    case 'graphicProjects':
      queries.content = TEAM_GRAPHIC_PROJECTS_QUERY;
      queries.count = TEAM_GRAPHIC_PROJECTS_COUNT_QUERY;
      queries.files = GRAPHIC_PROJECT_IMAGE_FILES_QUERY;
      queries.metaContent = GRAPHIC_PROJECTS_META_QUERY;
      queries.remove = DELETE_GRAPHIC_PROJECT_MUTATION;
      queries.status = UPDATE_GRAPHIC_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_GRAPHIC_PROJECT_MUTATION;

      return queries;
    case 'videoProjects':
      queries.content = TEAM_VIDEO_PROJECTS_QUERY;
      queries.count = TEAM_VIDEO_PROJECTS_COUNT_QUERY;
      queries.files = VIDEO_PROJECT_FILES_QUERY;
      queries.metaContent = VIDEO_PROJECTS_META_QUERY;
      queries.remove = DELETE_VIDEO_PROJECT_MUTATION;
      queries.status = UPDATE_VIDEO_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_VIDEO_PROJECT_MUTATION;

      return queries;
    case 'packages':
      queries.content = TEAM_PACKAGES_QUERY;
      queries.count = TEAM_PACKAGES_COUNT_QUERY;
      queries.files = PACKAGE_FILES_QUERY;
      queries.metaContent = PACKAGES_META_QUERY;
      queries.remove = DELETE_PACKAGE_MUTATION;
      queries.status = UPDATE_PACKAGE_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_PACKAGE_MUTATION;

      return queries;

    case 'playbooks':
      queries.content = TEAM_PLAYBOOKS_QUERY;
      queries.count = TEAM_PLAYBOOKS_COUNT_QUERY;
      // The following queries are placeholders and for the wrong content type
      // As such, these operations will not work properly
      queries.files = PACKAGE_FILES_QUERY;
      queries.metaContent = PACKAGES_META_QUERY;
      queries.remove = DELETE_PACKAGE_MUTATION;
      queries.status = UPDATE_PACKAGE_STATUS_MUTATION;
      queries.unpublish = UNPUBLISH_PACKAGE_MUTATION;

      return queries;
    default:
      return queries;
  }
};
