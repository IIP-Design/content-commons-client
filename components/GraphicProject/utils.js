import { getTransformedLangTaxArray, getCount } from 'lib/utils';

const structureLangObj = item => ( {
  language: {
    language_code: item.language.languageCode,
    text_direction: item.language.textDirection,
    locale: item.language.locale,
    display_name: item.language.displayName,
    native_name: item.language.nativeName,
  },
} );

export const normalizeGraphicProjectByAPI = ( { file, useGraphQl = false } ) => {
  const graphicObj = {
    id: file.id || '',
    type: 'graphic',
    site: file.site || '',
    title: file.title || '',
    projectType: file.projectType || '',
    published: file.published || '',
    modified: file.modified || '',
    owner: file.owner || '',
    desc: file.desc || '',
    descInternal: file.descInternal || '',
    copyright: file.copyright || '',
    images: file.images || [],
    supportFiles: file.supportFiles || [],
    categories: file.categories || '',
  };

  // Elastic API
  if ( !useGraphQl ) {
    const imagesWithFileNameProp = file.images.map( img => ( {
      ...img,
      filename: img.filename,
    } ) );

    const supportFilesWithFileNameProp = Array.isArray( file.supportFiles ) && file.supportFiles.map( supFile => {
      const filenameFromSrc = supFile.url.slice( supFile.url.lastIndexOf( '/' ) + 1 );

      return {
        ...supFile,
        filename: filenameFromSrc,
      };
    } );

    const esObj = {
      images: imagesWithFileNameProp || [],
      supportFiles: supportFilesWithFileNameProp || [],
    };

    return { ...graphicObj, ...esObj };
  }

  // GraphQL API
  if ( useGraphQl ) {
    const structuredImages = file.images.map( img => {
      const { dimensions } = img;

      return {
        ...img,
        ...structureLangObj( img ),
        url: img.signedUrl,
        width: dimensions.width,
        height: dimensions.height,
      };
    } );

    const structuredSupportFiles = Array.isArray( file.supportFiles ) && file.supportFiles.map( supFile => ( {
      ...supFile,
      ...structureLangObj( supFile ),
      srcUrl: supFile.signedUrl,
    } ) );

    const gqlObj = {
      alt: file.alt || '',
      projectType: file.type || '',
      published: file.publishedAt || '',
      modified: file.updatedAt || '',
      owner: file.team?.name || '',
      desc: file.descPublic || '',
      images: structuredImages || [],
      supportFiles: structuredSupportFiles || [],
      categories: getTransformedLangTaxArray( file.categories ) || [],
    };

    return { ...graphicObj, ...gqlObj };
  }

  return graphicObj;
};

// Use Twitter graphics as default for display otherwise whatever graphic image is available
export const filterGraphicImgs = images => {
  const containsTwitterImgs = images.some( img => img.social.includes( 'Twitter' ) );

  if ( containsTwitterImgs ) {
    return images.filter( img => img.social.includes( 'Twitter' ) );
  }

  return images;
};

export const getGraphicImgsBySocial = ( images, platform, useGraphQl = false ) => {
  if ( !getCount( images ) ) return [];

  const filteredImgs = images.reduce( ( allImgs, img ) => {
    let platformImgs = [];
    let condition = img.social.includes( platform );

    if ( useGraphQl ) {
      platformImgs = img.social.filter( s => s.name === platform );
      condition = getCount( platformImgs );
    }

    if ( condition ) {
      allImgs.push( img );
    }

    return allImgs;
  }, [] );

  return getCount( filteredImgs ) ? filteredImgs : images;
};
