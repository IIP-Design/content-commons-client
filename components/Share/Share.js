import React from 'react';
import PropTypes from 'prop-types';
import { List } from 'semantic-ui-react';
import { stringifyQueryString } from 'lib/browser';
import { contentRegExp, getVimeoId, getYouTubeId } from 'lib/utils';
import ClipboardCopy from '../ClipboardCopy';
import ShareButton from './ShareButton';

import './Share.scss';

const Share = props => {
  const {
    id, isPreview, site, language, title, link, type
  } = props;

  const internalOnly = type === 'document' || type === 'package';

  const queryStr = ( type === 'post' )
    ? stringifyQueryString( { id, site } )
    : stringifyQueryString( { id, site, language } );

  let directLink = link;
  let shareLink = link;

  // video, document, package types
  if ( !isPreview && type !== 'post' ) {
    directLink = `${window.location.protocol}//${window.location.host}/${type}?${queryStr}`;

    if ( link?.includes( 'youtu' ) ) {
      shareLink = `https://youtu.be/${getYouTubeId( link )}`;
    } else if ( link?.includes( 'vimeo' ) ) {
      shareLink = `https://vimeo.com/${getVimeoId( link )}`;
    }
  }

  // post type
  if ( contentRegExp( link ) && type === 'post' ) {
    directLink = `${window.location.protocol}//${window.location.host}/article?${queryStr}`;
  }

  // post, document, package types
  if ( type !== 'video' ) {
    shareLink = directLink;
  }

  const facebookURL = `https://www.facebook.com/sharer/sharer.php?u=${shareLink}`;
  const tweet = `https://twitter.com/intent/tweet?text=${title}&url=${shareLink}`;

  return (
    <div>
      { shareLink && !internalOnly && (
        <List className="share_list">
          <ShareButton
            url={ facebookURL }
            icon="facebook f"
            label="Share on Facebook"
            isPreview={ isPreview }
          />
          <ShareButton
            url={ tweet }
            icon="twitter"
            label="Share on Twitter"
            isPreview={ isPreview }
          />
        </List>
      ) }
      <ClipboardCopy
        label="Direct Link"
        copyItem={ directLink }
        isPreview={ isPreview }
      />
    </div>
  );
};

Share.defaultProps = {
  isPreview: false
};

Share.propTypes = {
  id: PropTypes.oneOfType( [
    PropTypes.number,
    PropTypes.string,
  ] ),
  isPreview: PropTypes.bool,
  site: PropTypes.string,
  language: PropTypes.string,
  link: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string
};

export default Share;
