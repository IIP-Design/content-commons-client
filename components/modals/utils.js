import GraphicProject from 'components/GraphicProject/GraphicProject';
import Post from 'components/Post/Post';
import Video from 'components/Video/Video';
import Document from 'components/Document/Document';
/**
 * Check content type of an item and passes in the correct component to the modal
 * @param {Object} item data from the selected item
 */
export const getModalContent = item => {
  const noContent = <div>No content currently available</div>;

  if ( item ) {
    switch ( item.type ) {
      case 'graphic':
        return <GraphicProject item={ item } displayAsModal />;

      case 'video':
        return <Video item={ item } />;

      case 'post':
        return <Post item={ item } />;

      case 'document':
        return <Document item={ item } displayAsModal />;

      default:
        return noContent;
    }
  }

  return noContent;
};
