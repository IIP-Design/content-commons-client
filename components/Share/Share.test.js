import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { getVimeoId, getYouTubeId } from 'lib/utils';
import Share from './Share';

jest.mock( 'next/config', () => () => ( {
  publicRuntimeConfig: {},
} ) );

const props = {
  id: 'c888',
  isPreview: false,
  site: '',
  language: 'en-us',
  link: 'https://player.vimeo.com/video/23729318',
  title: 'the title',
  type: 'video'
};

const Component = <Share { ...props } />;

describe( '<Share />', () => {
  it( 'renders without crashing', () => {
    const wrapper = mount( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'does not render the Facebook and Twitter share links if there is no link prop', () => {
    const wrapper = mount( Component );
    wrapper.setProps( { link: '' } );
    const list = wrapper.find( 'List.share_list' );

    expect( list.exists() ).toEqual( false );
  } );

  it( 'does not render the Facebook and Twitter share links if type is document', () => {
    const wrapper = mount( Component );
    wrapper.setProps( { type: 'document' } );
    const list = wrapper.find( 'List.share_list' );

    expect( list.exists() ).toEqual( false );
  } );

  it( 'does not render the Facebook and Twitter share links if type is package', () => {
    const wrapper = mount( Component );
    wrapper.setProps( { type: 'package' } );
    const list = wrapper.find( 'List.share_list' );

    expect( list.exists() ).toEqual( false );
  } );

  it( 'does not render the Facebook and Twitter share links if type is video', () => {
    const wrapper = mount( Component );
    wrapper.setProps( { type: 'video' } );
    const list = wrapper.find( 'List.share_list' );

    expect( list.exists() ).toEqual( false );
  } );

  it( 'passes the correct Facebook and Twitter share links if type is post and is from content.america.gov', () => {
    jest.resetModules();
    const wrapper = mount( Component );
    wrapper.setProps( { link: 'content.america.gov', type: 'post' } );

    const libBrowser = require( 'lib/browser' ); // eslint-disable-line
    libBrowser.stringifyQueryString = jest.fn( () => (
      `http://localhost/article?id=${props.id}&site=${props.site}`
    ) );

    const shareLink = libBrowser.stringifyQueryString();
    const facebookBtn = wrapper.find( 'ShareButton[icon="facebook f"]' );
    const twitterBtn = wrapper.find( 'ShareButton[icon="twitter"]' );
    const facebookShare = `https://www.facebook.com/sharer/sharer.php?u=${shareLink}`;
    const twitterShare = `https://twitter.com/intent/tweet?text=${props.title}&url=${shareLink}`;

    expect( wrapper.prop( 'link' ) ).toEqual( 'content.america.gov' );
    expect( wrapper.prop( 'type' ) ).toEqual( 'post' );
    expect( facebookBtn.prop( 'url' ) ).toEqual( facebookShare );
    expect( twitterBtn.prop( 'url' ) ).toEqual( twitterShare );
  } );

  it( 'passes the correct link to ClipboardCopy if type is video & !isPreview', () => {
    jest.resetModules();
    const wrapper = mount( Component );
    const clipboardCopy = wrapper.find( 'ClipboardCopy' );

    const libBrowser = require( 'lib/browser' ); // eslint-disable-line
    libBrowser.stringifyQueryString = jest.fn( () => (
      `http://localhost/video?id=${props.id}&site=${props.site}&language=${props.language}`
    ) );

    expect( clipboardCopy.prop( 'copyItem' ) )
      .toEqual( libBrowser.stringifyQueryString() );
  } );

  it( 'passes the placeholder text to ClipboardCopy if type is video & isPreview is true', () => {
    const wrapper = mount( Component );
    wrapper.setProps( {
      isPreview: true,
      link: 'The direct link to the project will appear here.'
    } );
    const clipboardCopy = wrapper.find( 'ClipboardCopy' );

    expect( clipboardCopy.prop( 'copyItem' ) )
      .toEqual( wrapper.prop( 'link' ) );
  } );

  it( 'passes the correct link to ClipboardCopy if type is post and is from content.america.gov', () => {
    jest.resetModules();
    const wrapper = mount( Component );
    wrapper.setProps( { link: 'content.america.gov', type: 'post' } );
    const clipboardCopy = wrapper.find( 'ClipboardCopy' );

    const libBrowser = require( 'lib/browser' ); // eslint-disable-line
    libBrowser.stringifyQueryString = jest.fn( () => (
      `http://localhost/article?id=${props.id}&site=${props.site}`
    ) );

    expect( wrapper.prop( 'link' ) ).toEqual( 'content.america.gov' );
    expect( wrapper.prop( 'type' ) ).toEqual( 'post' );
    expect( clipboardCopy.prop( 'copyItem' ) )
      .toEqual( libBrowser.stringifyQueryString() );
  } );

  it( 'passes the correct link to ClipboardCopy if type is neither post nor video', () => {
    jest.resetModules();
    const wrapper = mount( Component );
    wrapper.setProps( { type: 'image' } );
    const clipboardCopy = wrapper.find( 'ClipboardCopy' );

    const libBrowser = require( 'lib/browser' ); // eslint-disable-line
    libBrowser.stringifyQueryString = jest.fn( () => (
      `http://localhost/image?id=${props.id}&site=${props.site}&language=${props.language}`
    ) );

    expect( wrapper.prop( 'type' ) ).toEqual( 'image' );
    expect( clipboardCopy.prop( 'copyItem' ) )
      .toEqual(libBrowser.stringifyQueryString());
  } );
} );
