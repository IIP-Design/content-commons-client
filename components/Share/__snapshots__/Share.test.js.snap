// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Share /> renders without crashing 1`] = `
<Share
  id="c888"
  isPreview={false}
  language="en-us"
  link="https://player.vimeo.com/video/23729318"
  site=""
  title="the title"
  type="video"
>
  <div>
    <List
      className="share_list"
    >
      <div
        className="ui list share_list"
        role="list"
      >
        <ShareButton
          icon="facebook f"
          isPreview={false}
          label="Share on Facebook"
          url="https://www.facebook.com/sharer/sharer.php?u=https://vimeo.com/23729318"
        >
          <ListItem
            as="a"
            href="https://www.facebook.com/sharer/sharer.php?u=https://vimeo.com/23729318"
            onClick={[Function]}
            onKeyPress={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            target="_blank"
          >
            <a
              className="item"
              href="https://www.facebook.com/sharer/sharer.php?u=https://vimeo.com/23729318"
              onClick={[Function]}
              onKeyPress={[Function]}
              rel="noopener noreferrer"
              role="listitem"
              style={
                Object {
                  "cursor": "pointer",
                }
              }
              target="_blank"
            >
              <ListIcon
                name="facebook f"
              >
                <Icon
                  as="i"
                  className=""
                  name="facebook f"
                >
                  <i
                    aria-hidden="true"
                    className="facebook f icon"
                    onClick={[Function]}
                  />
                </Icon>
              </ListIcon>
              <ListContent>
                <div
                  className="content"
                >
                  Share on Facebook
                </div>
              </ListContent>
            </a>
          </ListItem>
        </ShareButton>
        <ShareButton
          icon="twitter"
          isPreview={false}
          label="Share on Twitter"
          url="https://twitter.com/intent/tweet?text=the title&url=https://vimeo.com/23729318"
        >
          <ListItem
            as="a"
            href="https://twitter.com/intent/tweet?text=the title&url=https://vimeo.com/23729318"
            onClick={[Function]}
            onKeyPress={[Function]}
            rel="noopener noreferrer"
            style={
              Object {
                "cursor": "pointer",
              }
            }
            target="_blank"
          >
            <a
              className="item"
              href="https://twitter.com/intent/tweet?text=the title&url=https://vimeo.com/23729318"
              onClick={[Function]}
              onKeyPress={[Function]}
              rel="noopener noreferrer"
              role="listitem"
              style={
                Object {
                  "cursor": "pointer",
                }
              }
              target="_blank"
            >
              <ListIcon
                name="twitter"
              >
                <Icon
                  as="i"
                  className=""
                  name="twitter"
                >
                  <i
                    aria-hidden="true"
                    className="twitter icon"
                    onClick={[Function]}
                  />
                </Icon>
              </ListIcon>
              <ListContent>
                <div
                  className="content"
                >
                  Share on Twitter
                </div>
              </ListContent>
            </a>
          </ListItem>
        </ShareButton>
      </div>
    </List>
    <ClipboardCopy
      copyItem="http://localhost/video?id=c888&site=&language=en-us"
      isPreview={false}
      label="Direct Link"
    >
      <div
        className="clipboardcopy_wrapper"
      >
        <div
          className="clipboardcopy"
        >
          <div
            className="clipboardcopy_label_wrapper"
          >
            <label
              className="clipboardcopy_label"
              htmlFor="clipboardcopy_item_text"
            >
              Direct Link
              <VisuallyHidden
                el="div"
              >
                <div
                  className="hide-visually"
                >
                   for this item
                </div>
              </VisuallyHidden>
            </label>
          </div>
          <div
            className="clipboardcopy_item"
          >
            <input
              className="clipboardcopy_item_text"
              defaultValue="http://localhost/video?id=c888&site=&language=en-us"
              disabled={false}
              id="clipboardcopy_item_text"
              onFocus={[Function]}
              readOnly={true}
              tabIndex="-1"
              type="text"
            />
            <Button
              as="button"
              className=""
              disabled={false}
              onClick={[Function]}
              primary={true}
            >
              <Ref
                innerRef={
                  Object {
                    "current": <button
                      class="ui primary button"
                    >
                      Copy
                    </button>,
                  }
                }
              >
                <RefFindNode
                  innerRef={
                    Object {
                      "current": <button
                        class="ui primary button"
                      >
                        Copy
                      </button>,
                    }
                  }
                >
                  <button
                    className="ui primary button"
                    onClick={[Function]}
                  >
                    Copy
                  </button>
                </RefFindNode>
              </Ref>
            </Button>
          </div>
        </div>
      </div>
    </ClipboardCopy>
  </div>
</Share>
`;
